### DISCUSSION TITLE
Unable to upload XML file

### DISCUSSION THREAD

**Discussion by aantgis** [INSPIRE-MIF/helpdesk-validator#988] [Q&A]
I have been trying to upload an XML file to the INSPIRE validator but I receive a generic error. The response I took from the support was that the XML was not encoded. I have tried several times to improve the XML but still with no luck. The XML is created by a Python script and below I have pasted a small demo:

<?xml version="1.0" encoding="UTF-8"?>
<InspireAddresses xmlns="<http://inspire.ec.europa.eu/schemas/ad/4.0>"
                  xmlns:gco="<http://www.isotc211.org/2005/gco>"
                  xmlns:gmd="<http://www.isotc211.org/2005/gmd>"
                  xmlns:gml="<http://www.opengis.net/gml/3.2>"
                  xmlns:gss="<http://www.isotc211.org/2005/gss>"
                  xmlns:gts="<http://www.isotc211.org/2005/gts>"
                  xmlns:xlink="<http://www.w3.org/1999/xlink>"
                  xmlns:xsi="<http://www.w3.org/2001/XMLSchema-instance>"
                  xsi:schemaLocation="<http://inspire.ec.europa.eu/schemas/ad/4.0> <http://inspire.ec.europa.eu/schemas/ad/4.0/Addresses.xsd>">
  <Address>
    <AddressType>Postal</AddressType>
    <Thoroughfare>
      <ThoroughfareName>Main Street</ThoroughfareName>
      <ThoroughfareNumber>123</ThoroughfareNumber>
      <ThoroughfareType>Street</ThoroughfareType>
    </Thoroughfare>
    <PostalIdentifier>10001</PostalIdentifier>
    <Municipality>New York</Municipality>
    <CountrySubdivision>New York State</CountrySubdivision>
    <PostCode>10001</PostCode>
  </Address>
  <Address>
    <AddressType>Delivery</AddressType>
    <Thoroughfare>
      <ThoroughfareName>Market Street</ThoroughfareName>
      <ThoroughfareNumber>456</ThoroughfareNumber>
      <ThoroughfareType>Avenue</ThoroughfareType>
    </Thoroughfare>
    <PostalIdentifier>94102</PostalIdentifier>
    <Municipality>San Francisco</Municipality>
    <CountrySubdivision>California</CountrySubdivision>
    <PostCode>94102</PostCode>
  </Address>
  <Address>
    <AddressType>Administrative</AddressType>
    <Thoroughfare>
      <ThoroughfareName>High Street</ThoroughfareName>
      <ThoroughfareNumber>789</ThoroughfareNumber>
      <ThoroughfareType>Road</ThoroughfareType>
    </Thoroughfare>
    <PostalIdentifier>SW1A 2AA</PostalIdentifier>
    <Municipality>London</Municipality>
    <CountrySubdivision>Greater London</CountrySubdivision>
    <PostCode>SW1A 2AA</PostCode>
  </Address>
  <Address>
    <AddressType>Postal</AddressType>
    <Thoroughfare>
      <ThoroughfareName>Gran Vía</ThoroughfareName>
      <ThoroughfareNumber>123</ThoroughfareNumber>
      <ThoroughfareType>Avenue</ThoroughfareType>
    </Thoroughfare>
    <PostalIdentifier>28013</PostalIdentifier>
    <Municipality>Madrid</Municipality>
    <CountrySubdivision>Community of Madrid</CountrySubdivision>
    <PostCode>28013</PostCode>
  </Address>
  <Address>
    <AddressType>Delivery</AddressType>
    <Thoroughfare>
      <ThoroughfareName>George Street</ThoroughfareName>
      <ThoroughfareNumber>456</ThoroughfareNumber>
      <ThoroughfareType>Street</ThoroughfareType>
    </Thoroughfare>
    <PostalIdentifier>2000</PostalIdentifier>
    <Municipality>Sydney</Municipality>
    <CountrySubdivision>New South Wales</CountrySubdivision>
    <PostCode>2000</PostCode>
  </Address>
</InspireAddresses>
[xml_test.zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/12117651/xml_test.zip)

**Comment by fabiovinci** [comment]
Dear @aantgis,

as I already replied to your similar question, the XML needs to conform to the [INSPIRE Addressed application schema](https://inspire.ec.europa.eu/schemas/ad/4.0/Addresses.xsd), please find attached a compliant gml file ([inspire-adreces-abella-conca-etrs89-geo.zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/12121670/inspire-adreces-abella-conca-etrs89-geo.zip)) discovered in the INSPIRE Geoportal.

You may find it useful to use a specific encoding tool instead of a Python script, e.g. [halestudio](https://wetransform.to/halestudio/).
